lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -1250,6 +1254,9 @@ RequirementCheckResult TypeChecker::checkGenericArguments(
       auto status = result.getStatus();
       switch (status) {
       case RequirementCheckResult::UnsatisfiedDependency:
+          if (listener)
+            listener->diagnoseUnsatisfiedRequirement(rawReq, firstType, secondType);
+          LLVM_FALLTHROUGH;
```

Comment:
This `diagnoseUnsatisfiedRequirement` isn't in the right place; it's `Failure` and `SubstitutionFailure` that would be diagnosed. Although at this point, it's dead code because you don't have a listener anywhere that diagnoses conformance failures, so I'd be okay with deleting this bit and adding a FIXME.